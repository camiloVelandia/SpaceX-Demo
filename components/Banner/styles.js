import styled from "styled-components";

export const Section = styled.section`
  height: 40vh;
  margin-top: 90px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: url(${props => props.backImage || "/images/rocket.jpg"});
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  position: relative;
  &::after {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.7);
    z-index: 0;
  }
  & h1 {
    z-index: 1;
    font-size: 48px;
    text-transform: uppercase;
  }
`;
