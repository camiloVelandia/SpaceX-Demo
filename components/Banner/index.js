import React from 'react';
import {Section} from './styles'

const Banner = (props) => {
  return (
    <Section>
      <h1> {props.title}</h1>
    </Section>
  );
};

export default Banner;