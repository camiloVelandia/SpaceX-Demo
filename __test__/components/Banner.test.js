import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import Banner from "../../components/Banner";

describe("<Banner/>", () => {
  const banner = mount(<Banner />);
  test("render del componente banner", () => {
    expect(banner.length).toEqual(1);
  });
});

describe("banner snaaapshot", () => {
  test("Comprobar ui del banner", () => {
    const banner = create(<Banner />);
    expect(banner.toJSON()).toMatchSnapshot();
  });
});
