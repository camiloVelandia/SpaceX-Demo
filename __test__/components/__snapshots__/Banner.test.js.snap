// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner snaaapshot Comprobar ui del banner 1`] = `
<section
  className="styles__Section-sc-15opjre-0 dIgbUX"
>
  <h1>
     
  </h1>
</section>
`;
